python/res/fm/rms/rms_run.py

```diff
@@ -153,22 +153,22 @@ def exec_rms(self, exec_env):
             args += ["-threads", str(self.config.threads)]
 
         if exec_env:
-            env = os.environ.copy()
-            for key, value in exec_env.items():
-                if value is None:
-                    env.pop(key)
-                    continue
-
-                value = str(value).strip()
-                if len(value) == 0:
-                    env.pop(key)
-                    continue
-
-                env[key] = value
-
-            os.execve( self.config.executable , args, env )
+            os.execve(self.config.executable, args, exec_env)
         else:
-            os.execv(self.config.executable, args )
-
-
+            os.execv(self.config.executable, args)
+
+
+    @staticmethod
+    def set_up_env(exec_env, env_extension):
+        """
+        This method takes in two dicts of environmental variables and sets up and
+        execute environment. If the values are empty the key will be popped.
```

Comment:
Please explain the difference between `exec_env` and `env_extension`. There is definitively someone in the future that will wonder what this does